Description: Switch SwiftShader to LLVM-16.0 for C++23 compatibility
 LLVM-10.0's APFloat.h has incomplete type issues when used with C++23's
 stricter template instantiation. Chromium 142.0.7444.162 ships both
 LLVM-10.0 and LLVM-16.0 in third_party/. LLVM-16.0 fixes the APFloat
 incomplete type error.
 .
 This is the same solution used by Debian Chromium package to fix
 SwiftShader compilation errors with C++23.
Author: Juan Manuel Méndez Rey <vejeta@gmail.com>
Bug-Debian: https://bugs.debian.org/915400
Forwarded: not-needed
Last-Update: 2025-01-17

diff --git a/third_party/swiftshader/src/Reactor/BUILD.gn b/third_party/swiftshader/src/Reactor/BUILD.gn
index 67dfeb0..dcb1361 100644
--- a/third_party/swiftshader/src/Reactor/BUILD.gn
+++ b/third_party/swiftshader/src/Reactor/BUILD.gn
@@ -307,7 +307,7 @@ if (supports_subzero) {

 if (supports_llvm) {
   swiftshader_source_set("swiftshader_llvm_reactor") {
-    llvm_dir = "../../third_party/llvm-10.0"
+    llvm_dir = "../../third_party/llvm-16.0"

     deps = [
       ":swiftshader_reactor_base",
